src/main/java/core/basesyntax/impl/AmountFromFileImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.AmountFromFile;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AmountFromFileImpl implements AmountFromFile {
+    private FruitShopDao fruitShopDao;
+
+    public AmountFromFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public List<Integer> getAmountInShop(List<String> fruits,
```

Comment:
The method name `getAmountInShop` does not accurately convey the action it performs. It not only retrieves amounts but also updates the `fruitShopDao`. Consider splitting the method into two: one for calculating the amounts and another for updating the storage.